### PR TITLE
Separates Functional Tests into Admin/Tenant

### PR DESCRIPTION
**Pull request by rbrady**
This patch changes the func tests to run separately for both admin
and tenant roles.  It follows the pattern done for the e2e testing
a short time ago.

The current run should fail to generate an error I plan to capture and place here in the discussion.  I have the change that will fix it in CI queued up and ready to push, but it may not be the right change long term.

### CODE REVIEW

**Comment by rbrady** [comment]
Here's the error when the 'devstack-system-admin' user is selected for src:

2020-09-29T15:28:29.5673914Z ok: [localhost] => (item={'auth_url': 'http://127.0.0.1/identity', 'password': 'devstack', 'project_domain_id': 'default', 'project_name': 'admin', 'user_domain_id': 'default', 'username': 'admin'}) => changed=false 
2020-09-29T15:28:29.5693057Z   ansible_loop_var: item
2020-09-29T15:28:29.5693687Z   item:
2020-09-29T15:28:29.5694095Z     auth_url: http://127.0.0.1/identity
2020-09-29T15:28:29.5694695Z     password: devstack
2020-09-29T15:28:29.5695078Z     project_domain_id: default
2020-09-29T15:28:29.5695872Z     project_name: admin
2020-09-29T15:28:29.5696411Z     user_domain_id: default
2020-09-29T15:28:29.5696785Z     username: admin
2020-09-29T15:28:30.2077792Z An exception occurred during task execution. To see the full traceback, use -vvv. The error was: keystoneauth1.exceptions.http.BadRequest: Expecting to find domain in user. The server could not comply with the request since it is either malformed or otherwise incorrect. The client is assumed to be in error. (HTTP 400) (Request-ID: req-0827233e-85eb-411b-977a-9175e406e1a0)
2020-09-29T15:28:30.2162771Z failed: [localhost] (item={'auth_url': 'http://127.0.0.1/identity', 'password': 'devstack', 'system_scope': 'all', 'username': 'admin'}) => changed=false 
2020-09-29T15:28:30.2180982Z   ansible_loop_var: item
2020-09-29T15:28:30.2181515Z   item:
2020-09-29T15:28:30.2182038Z     auth_url: http://127.0.0.1/identity
2020-09-29T15:28:30.2182931Z     password: devstack
2020-09-29T15:28:30.2183403Z     system_scope: all
2020-09-29T15:28:30.2183870Z     username: admin
2020-09-29T15:28:30.2184743Z   module_stderr: |-
2020-09-29T15:28:30.2185260Z     Traceback (most recent call last):
2020-09-29T15:28:30.2187917Z       File "/root/.ansible/tmp/ansible-tmp-1601393309.5721989-4134-184620573540416/AnsiballZ_os_keypair.py", line 102, in <module>
2020-09-29T15:28:30.2188829Z         _ansiballz_main()
2020-09-29T15:28:30.2190092Z       File "/root/.ansible/tmp/ansible-tmp-1601393309.5721989-4134-184620573540416/AnsiballZ_os_keypair.py", line 94, in _ansiballz_main
2020-09-29T15:28:30.2191354Z         invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
2020-09-29T15:28:30.2194172Z       File "/root/.ansible/tmp/ansible-tmp-1601393309.5721989-4134-184620573540416/AnsiballZ_os_keypair.py", line 40, in invoke_module
2020-09-29T15:28:30.2195801Z         runpy.run_module(mod_name='ansible.modules.cloud.openstack.os_keypair', init_globals=None, run_name='__main__', alter_sys=True)
2020-09-29T15:28:30.2196924Z       File "/usr/lib64/python3.7/runpy.py", line 205, in run_module
2020-09-29T15:28:30.2197452Z         return _run_module_code(code, init_globals, run_name, mod_spec)
2020-09-29T15:28:30.2198018Z       File "/usr/lib64/python3.7/runpy.py", line 96, in _run_module_code
2020-09-29T15:28:30.2198488Z         mod_name, mod_spec, pkg_name, script_name)
2020-09-29T15:28:30.2198963Z       File "/usr/lib64/python3.7/runpy.py", line 85, in _run_code
2020-09-29T15:28:30.2199372Z         exec(code, run_globals)
2020-09-29T15:28:30.2200097Z       File "/tmp/ansible_os_keypair_payload_dr07cohi/ansible_os_keypair_payload.zip/ansible/modules/cloud/openstack/os_keypair.py", line 156, in <module>
2020-09-29T15:28:30.2201165Z       File "/tmp/ansible_os_keypair_payload_dr07cohi/ansible_os_keypair_payload.zip/ansible/modules/cloud/openstack/os_keypair.py", line 123, in main
2020-09-29T15:28:30.2202351Z       File "/root/venv/lib64/python3.7/site-packages/openstack/cloud/_compute.py", line 416, in get_keypair
2020-09-29T15:28:30.2203232Z         return _utils._get_entity(self, 'keypair', name_or_id, filters)
2020-09-29T15:28:30.2204133Z       File "/root/venv/lib64/python3.7/site-packages/openstack/cloud/_utils.py", line 197, in _get_entity
2020-09-29T15:28:30.2204753Z         entities = search(name_or_id, filters, **kwargs)
2020-09-29T15:28:30.2205665Z       File "/root/venv/lib64/python3.7/site-packages/openstack/cloud/_compute.py", line 88, in search_keypairs
2020-09-29T15:28:30.2206350Z         filters=filters if isinstance(filters, dict) else None
2020-09-29T15:28:30.2207280Z       File "/root/venv/lib64/python3.7/site-packages/openstack/cloud/_compute.py", line 125, in list_keypairs
2020-09-29T15:28:30.2208027Z         return list(self.compute.keypairs(allow_unknown_params=True,
2020-09-29T15:28:30.2209032Z       File "/root/venv/lib64/python3.7/site-packages/openstack/service_description.py", line 87, in __get__
2020-09-29T15:28:30.2209717Z         proxy = self._make_proxy(instance)
2020-09-29T15:28:30.2210740Z       File "/root/venv/lib64/python3.7/site-packages/openstack/service_description.py", line 262, in _make_proxy
2020-09-29T15:28:30.2211358Z         found_version = temp_adapter.get_api_major_version()
2020-09-29T15:28:30.2212454Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/adapter.py", line 354, in get_api_major_version
2020-09-29T15:28:30.2213224Z         return self.session.get_api_major_version(auth or self.auth, **kwargs)
2020-09-29T15:28:30.2214268Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/session.py", line 1266, in get_api_major_version
2020-09-29T15:28:30.2214925Z         return auth.get_api_major_version(self, **kwargs)
2020-09-29T15:28:30.2216122Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/identity/base.py", line 500, in get_api_major_version
2020-09-29T15:28:30.2216922Z         data = get_endpoint_data(discover_versions=discover_versions)
2020-09-29T15:28:30.2218905Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/identity/base.py", line 271, in get_endpoint_data
2020-09-29T15:28:30.2219739Z         service_catalog = self.get_access(session).service_catalog
2020-09-29T15:28:30.2220854Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/identity/base.py", line 134, in get_access
2020-09-29T15:28:30.2221541Z         self.auth_ref = self.get_auth_ref(session)
2020-09-29T15:28:30.2222550Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/identity/generic/base.py", line 208, in get_auth_ref
2020-09-29T15:28:30.2223310Z         return self._plugin.get_auth_ref(session, **kwargs)
2020-09-29T15:28:30.2224318Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/identity/v3/base.py", line 188, in get_auth_ref
2020-09-29T15:28:30.2225043Z         authenticated=False, log=False, **rkwargs)
2020-09-29T15:28:30.2225991Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/session.py", line 1139, in post
2020-09-29T15:28:30.2227010Z         return self.request(url, 'POST', **kwargs)
2020-09-29T15:28:30.2227960Z       File "/root/venv/lib64/python3.7/site-packages/keystoneauth1/session.py", line 976, in request
2020-09-29T15:28:30.2228744Z         raise exceptions.from_response(resp, method, url)
2020-09-29T15:28:30.2230744Z     keystoneauth1.exceptions.http.BadRequest: Expecting to find domain in user. The server could not comply with the request since it is either malformed or otherwise incorrect. The client is assumed to be in error. (HTTP 400) (Request-ID: req-0827233e-85eb-411b-977a-9175e406e1a0)
2020-09-29T15:28:30.2232808Z   module_stdout: ''
2020-09-29T15:28:30.2233197Z   msg: |-
2020-09-29T15:28:30.2233466Z     MODULE FAILURE
2020-09-29T15:28:30.2233806Z     See stdout/stderr for the exact error
2020-09-29T15:28:30.2234090Z   rc: 1
2020-09-29T15:28:30.2245357Z 
2020-09-29T15:28:30.2245791Z PLAY RECAP *********************************************************************
2020-09-29T15:28:30.2246321Z localhost                  : ok=6    changed=2    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
2020-09-29T15:28:30.2249211Z 
2020-09-29T15:28:30.2936499Z make: *** [Makefile:81: test-func-admin] Error 2

File: .github/workflows/consistency-functional.yml [hunk 1]
```diff
@@ -268,6 +268,11 @@ jobs:
               --src devstack \
               --dst devstack-alt \
               > tests/auth_tenant.yml && \
```
Comment (ccamacho) [inline .github/workflows/consistency-functional.yml:270]: On each PR we will run both admin and tenant, maybe we can split the job in two with a dependence any way this looks like a nice improvement

**Review by ccamacho — APPROVED** [review]

File: .github/workflows/consistency-functional.yml (hunk 1 above)
Comment (jistr) [reply .github/workflows/consistency-functional.yml:270]: +1

**Review by jistr — APPROVED** [review]
Great, thanks!

**Pull request merged**